src/lazy/text/matched.rs

```diff
@@ -107,3 +111,46 @@ impl MatchedInt {
         Ok(int)
     }
 }
+
+/// A partially parsed Ion float.
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub(crate) enum MatchedFloat {
+    /// `+inf`
+    PositiveInfinity,
+    /// `-inf`
+    NegativeInfinity,
+    /// `nan`
+    NotANumber,
+    /// Any numeric float value
+    Numeric,
+}
```

Comment:
🗺️ PR #609 introduced an `EncodedValue` struct that stores information like offset and length, but which also has a `MatchedValue` enum field. This is one of the variants of `MatchedValue`, and remembers the kind of `float` we encountered in case we read it later. The offset and length will be available in the parent data structure.